### PR TITLE
Fix system and template generator for Espaloma 0.3.*

### LINKED ISSUE #245: [Enhancement] Add options to EspalomaTemplateGenerator to deploy different espaloma models
The current implementation for EspalomaTemplateGenerator only accepts espaloma models that were jointly trained with charge losses which limits the use of different charge methods. Additionally, when the openmm system is created ([here](https://github.com/openmm/openmmforcefields/blob/main/openmmforcefields/generators/template_generators.py#1710)), there is no way to specify the forcefield used assign nonbonded parameters.

I think it would be great if we could have options to specify the charge method and the forcefield used to assign the nonbonded parameters.

### LINKED ISSUE #263: EspalomaTemplateGenerator cannot properly load espaloma models trained with dropouts and batch normalization
We need to call model.eval() after loading the espaloma model in [generators/template_generators.py#L1580](https://github.com/openmm/openmmforcefields/blob/main/openmmforcefields/generators/template_generators.py#L1580) when loading a model trained with dropouts and batch normalization. It will yield inconsistent inference results if model.eval() is not called.

Reference:
https://pytorch.org/tutorials/beginner/saving_loading_models.html

### LINKED ISSUE #282: Check if partial charges are preserved when using `from-molecule` charge method in `EspalomaTemplateGenerator`
Good catch! We need to make sure if this is the case because we want users to be able to use charges from other methods/models combined with espaloma. I'll raise an issue about this so we can keep track of this.

_Originally posted by @ijpulidos in https://github.com/openmm/openmmforcefields/pull/271#discussion_r1231611892_

### PR DESCRIPTION
**Pull request by ijpulidos**
Created a new PR with these changes such that we can use the CI secrets from our repo. Supersedes #271 

These changes add `template_generator_kwargs` to the system generators.

Resolves #245 . Resolves #263 . Resolves #282 

## TODO

- [x] add tests

### CODE REVIEW

File: openmmforcefields/generators/template_generators.py [hunk 1]
```diff
@@ -1558,6 +1567,12 @@ def __init__(self, molecules=None, cache=None, forcefield=None, model_cache_path
         >>> espaloma_generator = EspalomaTemplateGenerator(cache='smirnoff.json', forcefield='espaloma-0.2.2')
 
         Newly parameterized molecules will be written to the cache, saving time next time!
+
+        You can also pass a template_generator_kwargs to specify the reference_forcefield and/or charge_method in EspalomaTemplateGenerator:
+
+        >>> template_generator_kwargs = {"reference_forcefield": "openff_unconstrained-2.0.0", "charge_method": "nn"}
+        >>> espaloma_generator = EspalomaTemplateGenerator(cache='smirnoff.json', forcefield='espaloma-0.2.2',
```
Comment (jchodera) [inline openmmforcefields/generators/template_generators.py:1574]: Note to self: We need to update the docstrings.

File: devtools/conda-envs/test_env.yaml [hunk 2]
```diff
@@ -2,7 +2,6 @@ name: openmmforcefields-test
 
 channels:
   - conda-forge
-  - dglteam # espaloma; TODO: Remove this once espaloma is fully on conda-forge
   - defaults
```
[the remark this answers is not in the document]
Comment (mikemhenry) [reply devtools/conda-envs/test_env.yaml:5]: I'll take a quick pass at removing this

**Comment by jchodera** [comment]
@mikemhenry @ijpulidos : I've finished cleaning up the docs and running local tests.
After this passes tests, can you merge and cut a new release?
